### ISSUE TITLE
Locations not listed in user self-settings

### ISSUE THREAD

**Issue by NebelKreis** [grokability/snipe-it#13269]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

We want the users to be able to change their location themselves. The problem is that if they try it with the normal worker group, then no locations are listed and the named errors occur. 
 
The worker group has only the ability to self edit their location.  
 
Even if we don’t use a group and manually assign the permission to one user, the problem occurs. 
 
We also tried granting them the permission Locations: View but the problem persists. If they go to the location settings, every location will be listed successfully.

### Reproduction steps

1. Create a user only with the permission „_Self: edit location_“.
2. Login as this user.
3. Now try changing the location.
4. The list should say „_The results could not be loaded._“
5. Try searching for existing locations and again, the list should say „_The results could not be loaded._“.

### Expected behavior

It should list every location that exists.

### Snipe-IT Version

6.1.2

### Operating System

Ubuntu Server 20.04

### Web Server

Apache/2.4.41

### PHP Version

8.1.20

### Operating System

Windows

### Browser

Google Chrome

### Error messages

```shell
Chrome:
Failed to load resource: the server responded with a status of 403 (Forbidden)
/api/v1/locations/selectlist?page=1:1     

 
GET https://<Link>/api/v1/locations/selectlist?page=1 403 (Forbidden)
send @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
ajax @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
transport @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
o @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.i.query @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
(anonymous) @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.i.invoke @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.i.trigger @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.r.trigger @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.r.open @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.r.toggleDropdown @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
(anonymous) @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.i.invoke @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
e.define.i.trigger @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
(anonymous) @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
dispatch @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
m.handle @ all.js?id=fc2a78ab0c9ec6ce0961990388a1b23f:1
 
 
Laravel:
No errors that are relevant for this problem.
```

### Additional context

The problem has persisted since the first installation. Upgrading to `6.1.2` didn’t fix the problem.
The installation was done manually.

**Comment by Dad-Ju** [comment]
Hey,
we got into it and we found out that the user is missing the `locations.view.selectlist` permission, this should apply when the user gets the `self.edit_location` or not?

Adding the Locations View Permissions doesnt let the user see the dropdown (because view.selectlist is still missing)

This seams to be a bug in the code!

![image](https://github.com/snipe/snipe-it/assets/26618439/ab977e4d-7493-4405-bc1c-a6a0a8d809b9)

**Comment by marcusmoore** [comment]
Hi there. I just opened a PR for this (#13310). Follow along to see if/when it gets merged 😄

**Comment by Dad-Ju** [comment]
Thanks alot for the fast "Fix" @marcusmoore 👍🏼

**Comment by marcusmoore** [comment]
Fixed in #13310
